azure/vnet/variables.tf

```diff
@@ -0,0 +1,270 @@
+  type        = string
+  default     = null
+  description = "(Optional) Specifies the Edge Zone within the Azure Region where this Virtual Network should exist. Changing this forces a new Virtual Network to be created."
+}
+
+variable "flow_timeout_in_minutes" {
+  type        = number
+  default     = null
+  description = "(Optional) The flow timeout in minutes for the Virtual Network, which is used to enable connection tracking for intra-VM flows. Possible values are between 4 and 30 minutes."
+}
+
+variable "ddos_protection_plan" {
+  description = "DDoS protection plan settings"
+  type        = any
+  default     = null
+}
+
+variable "bgp_community" {
+  type        = string
+  default     = null
+  description = "(Optional) The BGP community attribute in format <as-number>:<community-value>."
+}
+
+
+# ----------------------------------------------------------------------------------------------------------------------
+# Subnet Variables
+# ----------------------------------------------------------------------------------------------------------------------
+variable "subnets" {
+  type = list(object({
+    prefix            = string
+    address_newbits   = number
+    address_netnum    = number
+    service_endpoints = list(string)
+    delegation = object({
+      name = string
+      service_delegation = object({
+        name    = string
+        actions = list(string)
+      })
+    })
+    private_link_service_network_policies_enabled = bool
+    private_endpoint_network_policies_enabled     = bool
+    nsg_rules = object({
+      deny_all_inbound                  = bool
+      http_inbound_allowed              = bool
+      https_inbound_allowed             = bool
+      ssh_inbound_allowed               = bool
+      rdp_inbound_allowed               = bool
+      winrm_inbound_allowed             = bool
+      application_gateway_rules_enabled = bool
+      load_balancer_rules_enabled       = bool
+      nfs_inbound_allowed               = bool
+      cifs_inbound_allowed              = bool
+      allowed_http_source               = any
+      allowed_https_source              = any
+      allowed_ssh_source                = any
+      allowed_rdp_source                = any
+      allowed_winrm_source              = any
+      allowed_nfs_source                = any
+      allowed_cifs_source               = any
+      custom_security_rules = list(object({
+        name                         = string
+        access                       = string
+        direction                    = string
+        priority                     = number
+        protocol                     = string
+        source_port_range            = string
+        destination_port_range       = string
+        source_address_prefix        = string
+        destination_address_prefix   = string
+        source_address_prefixes      = list(string)
+        destination_address_prefixes = list(string)
+      }))
+    })
+  }))
+  default     = []
```

Comment:
Should we set a default for this variable so that if an end user downloads and attempts to use this module without passing in a subnets variable, it might create 1 private subnet and 1 public subnet?

Reply:
@TysonBeck stated;
 defaults which is a really hard one as you never want overlapping cidrs in a network but vnet does have a default... I think it is best to leave it blank which forces the entree and thought of the value which can reduce issues with quick/test deployments